src/Raven.Server/Documents/Indexes/IndexStorage.cs

```diff
@@ -192,8 +192,14 @@ void PersistSearchEngine(Tree configurationTree)
                     string configurationKey = nameof(SearchEngineType);
                     string configurationName = _index.Type.IsAuto() ? RavenConfiguration.GetKey(x => x.Indexing.AutoIndexingEngineType) : RavenConfiguration.GetKey(x => x.Indexing.StaticIndexingEngineType);
 
-                    SearchEngineType defaultEngineType =
-                        _index.Type.IsAuto() ? _index.Configuration.AutoIndexingEngineType : _index.Configuration.StaticIndexingEngineType;
+                    SearchEngineType defaultEngineType = _index.Type.IsAuto() switch
+                    {
+                        // We only support Vectors in Corax, so if an auto-index is using it, let's already set it up as such, regardless
+                        // of what type of default storage engine is configured.
+                        true when _index.Definition.IndexFields.Any(x=> x.Value.Vector != null) => SearchEngineType.Corax,
```

Comment:
let's talk about this next week

Reply:
ping

Reply:
We know what to do, but it was not implemented yet.

1. If default auto index engine is corax => no problem
2. if default auto index is lucene then new auto index should be created based on corax, but we need to make adjustments in auto index matcher to avoid accidental sudden switch of queries from one indexing engine to another